lib/livebook/session/recently_opened.ex

```diff
@@ -0,0 +1,71 @@
+defmodule Livebook.Session.RecentlyOpened do
+  @moduledoc false
+
+  use GenServer
+
+  alias Livebook.Session
+  alias Livebook.Users.User
+  alias Livebook.Storage
+
+  @namespace :recently_opened_sessions
+  @max_size 10
+
+  def start_link(arg) do
+    GenServer.start_link(__MODULE__, arg, name: __MODULE__)
+  end
+
+  def list_sessions(user) do
+    GenServer.call(__MODULE__, {:list_sessions, user})
+  end
+
+  @spec save_session(User.t(), Session.t()) :: :ok
+  def save_session(user, session) do
+    GenServer.cast(__MODULE__, {:save_session, user, session})
+  end
+
+  @spec delete_session(User.t(), Session.id()) :: :ok
+  def delete_session(user, session_id) do
+    GenServer.cast(__MODULE__, {:delete_session, user, session_id})
+  end
+
+  @impl true
+  def init(_) do
+    state =
+      Storage.all(@namespace)
+      |> Map.new(&{&1.id, &1.sessions})
+
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_call({:list_sessions, user}, _from, state) do
+    {:reply, Map.get(state, user.id, []), state}
+  end
+
+  @impl true
+  def handle_cast({:save_session, user, session}, state) do
+    new_state =
+      Map.update(state, user.id, [session], fn sessions ->
+        sessions
+        |> Kernel.--([session])
+        |> List.insert_at(0, session)
```

Comment:
Sessions are temporary, once Livebook is restarted all of them are gone. What we actually want to store is the notebook file locations. This way when Livebook is restarted we can easily reopen a notebook we recently worked on.

Reply:
In this context, one case to consider is when someone changes a session file, we need to update it here accordingly. I think we can store session_id only in the state and also subscribe to changes via `Livebook.Sessions.subscribe()`, then we can detect when a particular file changes.

Reply:
@jonatanklosko I would rather make it so if we change the file, we explicitly notify the SessionManager :)

Reply:
@josevalim we already broadcast updates to the session struct, which includes the file, so it feels redundant to broadcast both events separately, no?

Reply:
broadcasts are more like for "fire and forget" functionality. In this case, I'd say it is important to guarantee SessionManager picks it up, so I think a call would be a more appropriate. But I only have slightly strong feelings about it. :)

Reply:
Ahh, I didn't get it initially, you mean we call it directly, sounds good to me!

Reply:
So, is it correct that `RecentlyOpened` GenServer only stores session IDs?
So what is Session Manager?
And where do I save the file location for the recently opened sessions?

Reply:
We only need a GenServer which will  keep a list of recently opened files and persist it to disk using Storage.

Reply:
I got it.
Would you please review my commit?

Reply:
@ByeongUkChoi we are still storing sessions, but what we want to store is the notebook file :) Then we also need to notify the genserver when a session file changes (from `Session`).

Let's rename `RecentlyOpened` to `SessionManager`, as we may want to use it for other things in the future.

Reply:
rename to `SessionManager` !

Reply:
Cool, next we need to change it to store notebook files and not sessions :)

Reply:
So, SessionManager GenServer store list of Notebook struct? or list of Notebook path?

Reply:
Notebook paths :)